lxd/storage_volumes.go

```diff
@@ -1852,6 +1856,63 @@ func storagePoolVolumeDelete(d *Daemon, r *http.Request) response.Response {
 	return response.EmptySyncResponse
 }
 
+func createStoragePoolVolumeFromISO(s *state.State, r *http.Request, requestProjectName string, projectName string, data io.Reader, pool string, volName string) response.Response {
+	revert := revert.New()
+	defer revert.Fail()
+
+	if volName == "" {
+		return response.BadRequest(fmt.Errorf("Missing volume name"))
+	}
+
+	// Create temporary file to store uploaded ISO data.
+	isoFile, err := os.CreateTemp(shared.VarPath("backups"), fmt.Sprintf("%s_", backup.WorkingDirPrefix))
+	if err != nil {
+		return response.InternalError(err)
+	}
+
+	defer func() { _ = os.Remove(isoFile.Name()) }()
+	revert.Add(func() { _ = isoFile.Close() })
+
+	// Stream uploaded ISO data into temporary file.
+	size, err := io.Copy(isoFile, data)
```

Comment:
We might not need 2 full copies for ISOs. It makes sense for backups because we need to seek. But for ISOs we don't do that. In a previous iteration of this PR, where we checked the ISO header, we would've needed 2 full copies.

Reply:
If we don't have 2 full copies though we need to get the file size from somewhere else. The client would need to set the `Content-Length` header.

Reply:
I did some testing, and I get the following error:
```
Error: Failed creating custom volume from ISO: Failed creating volume: http: invalid Read on closed Body
```

That's probably the reason for using a temporary file.